articlequality/feature_lists/enwiki.py

```diff
@@ -58,17 +60,40 @@
     name="enwiki.revision.non_cite_templates"
 )
 
+revision = Revision(
+    "enwiki.revision.revision",
+    wikitext.revision.datasources,
+)
+
 # Links
 category_links = wikitext.revision.wikilink_titles_matching(
     r"Category\:", name="enwiki.revision.category_links")
-image_links = wikitext.revision.wikilink_titles_matching(
-    r"File|Image\:", name="enwiki.revision.image_links")
+
+image_templates_str = revision.templates_str_matching(
+    r"{{(multiple image|image array|gallery|photomontage)",
+    name='enwiki.revision.image_templates_str')
+image_tags_str = revision.tags_str_matching(
+    r"<(gallery|imagemap)",
+    name='enwiki.revision.image_tags_str')
+image_titles = wikitext.revision.wikilink_titles_matching(
+    r"File|Image\:",
+    name='enwiki.revision.image_titles')
+image_template_names = wikitext.revision.template_names_matching(
+    r"((Wide|Tall|scalable) image)|Panorama|Panorama 2",
+    name='enwiki.revision.image_template_names')
+
+image_links = Feature(
+    "enwiki.revision.image_links",
+    lambda a, b, c, d: len(
```

Comment:
For weird reasons, lambdas don't work as `process` functions.  You'll need to define the function as a first order.  

Also, it would be better to define some features for images_in_templates and gallery_images and then add them together like this:

```
all_images = image_links + image_templates + images_in_templates + gallery_images
```

Reply:
Thanks for the feedback. The `gallery_images` will be for both the tag and template gallery types right? As we can have <gallery> and {{gallery...